### PR TITLE
Re-use param index 3 for Fence Action

### PR DESCRIPTION
**Pull request by Igor-Misic**
This PR is connected to: https://github.com/mavlink/mavlink/issues/2043

It adds the option to use `index param 3` for sending fence action, so each geofence can have its breach action. 

Feasibility development was done and with these changes, we can have multiple steps of breaching actions. One useful example is a case where the vehicle flies uncontrollably outside the last fence we can terminate the vehicle, to prevent the vehicle from entering a forbidden area, etc.
image

### CODE REVIEW

File: message_definitions/v1.0/common.xml [hunk 1]
```diff
@@ -2333,7 +2333,7 @@
         </description>
         <param index="1" label="Vertex Count" minValue="3" increment="1">Polygon vertex count</param>
         <param index="2" label="Inclusion Group" minValue="0" increment="1">Vehicle must be inside ALL inclusion zones in a single group, vehicle must be inside at least one group, must be the same for all points in each polygon</param>
-        <param index="3">Reserved</param>
+        <param index="3" label="Fence Action" minValue="0" increment="1">Action the vehicle will take when the fence is breached.</param>
```
Comment (hamishwillee) [inline message_definitions/v1.0/common.xml:2336]: 1. Should be an enum (true for all cases)
2. Make it clear that in polygon all values should be the same as each other, or will be ignored.
3. Allow for the 0 case to mean system default.

**Comment by hamishwillee** [comment]
@Igor-Misic We discussed this in [20231122](https://github.com/mavlink/mavlink/wiki/20231122-Dev-Meeting) dev call. Essentially we agree in principle but we won't merge until there is a prototype that demonstrates this does indeed do what is needed (i.e. we want to be sure this is really going in before we update common.xml).

Also make sure the existing comments such as https://github.com/mavlink/mavlink/pull/2048#discussion_r1385965434 are addressed.

**Comment by Igor-Misic** [comment]
Thanks for talking about this topic in the discussion. At the moment I don't have the resources to push this upstream, so I'll put this in a pause then. Anyhow I think it was good progress for anyone who will want to implement multiple geofence actions.

**Comment by hamishwillee** [comment]
Thanks @Igor-Misic - that's disappointing, would have been cool. But closing it is the right thing to do if we can't get it in.

**Pull request closed without merging**